packages/keymanager-server/src/impl.ts

```diff
@@ -172,14 +169,14 @@ export class KeymanagerApi implements Api {
         // Remove from Sync committee duties
         // Remove from indices
         this.validator.removeDutiesForKey(pubkeyHex);
-        const keystorePathInfo = this.getKeystorePathInfoForKey(pubkeyHex);
+
+        const keystoreFilepath = this.getKeystoreFilepath(pubkeyHex);
+
         // Remove key from persistent storage
-        for (const keystoreFile of await fs.promises.readdir(this.importKeystoresPath)) {
-          if (keystoreFile.indexOf(pubkeyHex) !== -1) {
-            await fs.promises.unlink(keystorePathInfo.keystoreFilePath);
-            await fs.promises.unlink(keystorePathInfo.lockFilePath);
-          }
-        }
```

Comment:
@dadepo Why was it necessary to loop over `importKeystoresPath`? The file paths should be predictable

Reply:
I think the loop was from an initial implementation that did not use a stable path/filename. So indeed a loop is not necessary